app/models/work.rb

```diff
@@ -249,6 +203,36 @@ def expire_caches
     end
   end
 
+  def expire_pseud(pseud)
+    CacheMaster.record(self.id, 'pseud', pseud.id)
+    CacheMaster.record(self.id, 'user', pseud.user_id)
+  end
+
+  # When works are done being reindexed, expire the appropriate caches
+  def self.successful_reindex(ids)
+    CacheMaster.expire_caches(ids)
+    tag_ids = FilterTagging.where(filterable_id: ids, filterable_type: 'Work').
+                            group(:filter_id).
+                            value_of(:filter_id)
+
+    collection_ids = CollectionItem.where(item_id: ids, item_type: 'Work').
+                                    group(:collection_id).
+                                    value_of(:collection_id)
+
+    pseuds = Pseud.select("pseuds.id, pseuds.user_id").
+                    joins(:creatorships).
+                    where(creatorships: {
+                      creation_id: ids, 
+                      creation_type: 'Work'
+                      }
+                    )
+
+    pseuds.each { |p| p.update_works_index_timestamp! }
```

Comment:
Pass `&:update_works_index_timestamp!` as an argument to `each` instead of a block.